### PR TITLE
packages: reduce use of io.ReadAll for npm packages

### PR DESCRIPTION
**Pull request by Strum355**
Continuation of https://github.com/sourcegraph/sourcegraph/pull/44497 but for npm packages

## Test plan

Unit test and prod :trollface:

### CODE REVIEW

**Review by eseliger — APPROVED** [review]

File: internal/extsvc/npm/npm.go [hunk 1]
```diff
@@ -169,18 +179,20 @@ func (client *HTTPClient) GetDependencyInfo(ctx context.Context, dep *reposource
...
 	}
+
 	return &info, nil
 }
 
 func (client *HTTPClient) FetchTarball(ctx context.Context, dep *reposource.NpmVersionedPackage) (io.ReadCloser, error) {
 	if dep.TarballURL == "" {
 		return nil, errors.New("empty TarballURL")
 	}
+
```
Comment (varungandhi-src) [inline internal/extsvc/npm/npm.go:196]: Q: Why are there so many whitespace changes in this patch? There doesn't seem to be any grouping, having extra lines around each branch makes the code annoying to read. 😕 This function literally has one branch and two return statements? Do we need the extra line in-between?

File: internal/extsvc/npm/npm.go [hunk 2]
```diff
@@ -149,17 +158,18 @@ func (client *HTTPClient) makeGetRequest(ctx context.Context, url string) (io.Re
...
-	var bodyBuffer bytes.Buffer
-	if _, err := io.Copy(&bodyBuffer, resp.Body); err != nil {
-		return nil, err
-	}
 	if resp.StatusCode >= 400 {
+		defer resp.Body.Close()
+
+		var bodyBuffer bytes.Buffer
+		if _, err := io.Copy(&bodyBuffer, resp.Body); err != nil {
+			return nil, npmError{resp.StatusCode, errors.Newf("error reading non-200 response body: %s", bodyBuffer.String())}
```
Comment (varungandhi-src) [inline internal/extsvc/npm/npm.go:167]: nit: There will already be an 'error:'/'warn:' prefix in any log message, this should probably be worded 'failed to read' or similar.

Comment (Strum355) [reply internal/extsvc/npm/npm.go:167]: I think i made this one inconsistent with the other PRs, I'll look over em again to see what it was

File: internal/extsvc/npm/npm.go (hunk 1 above)
Comment (Strum355) [reply internal/extsvc/npm/npm.go:196]: Theres a fairly even split between newlines after branches vs none, youre right about these short ones though, Ill go ahead with something more sensible

**Review by varungandhi-src — APPROVED** [review]

**Pull request merged**